src/models/client.ts

```diff
@@ -209,30 +219,65 @@ export class ClientEnvironment {
     await this.network.clearBrowserCache();
   }
 
-  public async navigate(arg?: string | NavigateOptions): Promise<NavigateResult> {
-    const targetUrl = arg ? typeof arg === 'string' ? arg : arg.targetUrl : null;
-    const url = targetUrl ? this.getAbsoluteUrl(targetUrl) : this.rootUrl;
-    const waitForLoad = typeof arg === 'object' && arg.waitForLoad;
-
-    const tree = await this.page.getFrameTree();
-    const frameId = tree.frameTree.frame.id;
+  private async buildCapturedResponse(response: Network.ResponseReceivedParameters, shouldAddBodies: boolean) {
+    const capturedResponse = {responseMeta: response.response} as CapturedResponse;
```

Comment:
Why does this need a cast? Does this not work?
```
const capturedResponse: CapturedResponse = { responseMeta: response.response };
```

Reply:
Yeah, "const capturedResponse: CapturedResponse" works fine. I'm just so unused to having to type in an assignment of a new object in ts that I fell back to the cast syntax by default.